src/test/java/edu/ucsb/cs156/happiercows/controllers/StudentsControllerTests.java

```diff
@@ -0,0 +1,302 @@
+                                .fmName("AlecJ2")
+                                .email("alecsong2@ucsb.edu")
+                                .perm("12345672")
+                                .courseId((long)1562)
+                                .build();
+
+                String requestBody = mapper.writeValueAsString(StudentsEdited);
+
+                when(StudentsRepository.findById(eq(67L))).thenReturn(Optional.of(StudentsOrig));
+
+                // act
+                MvcResult response = mockMvc.perform(
+                                put("/api/Students?id=67")
+                                                .contentType(MediaType.APPLICATION_JSON)
+                                                .characterEncoding("utf-8")
+                                                .content(requestBody)
+                                                .with(csrf()))
+                                .andExpect(status().isOk()).andReturn();
+
+                // assert
+                verify(StudentsRepository, times(1)).findById(67L);
+                verify(StudentsRepository, times(1)).save(StudentsEdited); // should be saved with correct user
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals(requestBody, responseString);
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void admin_cannot_edit_Students_that_does_not_exist() throws Exception {
+
+                Students StudentsEdited = Students.builder()
+                                .lName("Song")
+                                .fmName("AlecJ")
+                                .email("alecsong@ucsb.edu")
+                                .perm("1234567")
+                                .courseId((long)156)
+                                .build();
+
+                String requestBody = mapper.writeValueAsString(StudentsEdited);
+
+                when(StudentsRepository.findById(eq(67L))).thenReturn(Optional.empty());
+
+                // act
+                MvcResult response = mockMvc.perform(
+                                put("/api/Students?id=67")
+                                                .contentType(MediaType.APPLICATION_JSON)
+                                                .characterEncoding("utf-8")
+                                                .content(requestBody)
+                                                .with(csrf()))
+                                .andExpect(status().isNotFound()).andReturn();
+
+                // assert
+                verify(StudentsRepository, times(1)).findById(67L);
+                Map<String, Object> json = responseToJson(response);
+                assertEquals("Students with id 67 not found", json.get("message"));
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void admin_can_delete_a_Students() throws Exception {
```

Comment:
```suggestion
        public void admin_can_delete_a_student() throws Exception {
```

This sort of thing "doesn't matter" from a functionality standpoint, but in terms of projecting "professionalism" and "attention to detail" it really does.  It's the kind of things that stands out, for example, if you are doing coding as part of a job interview—which candidates take the time to name things carefully and consistently?  Which leave small typos and inconsistencies?